### ISSUE TITLE
Panic in leader election, close of closed channel

### ISSUE THREAD

**Issue by therc** [kubernetes/kubernetes#26782]
Built from HEAD. Edited for conciseness.

```
m0 kube-scheduler: E0603 14:41:02.243120   29837 event.go:257] Could not construct reference to: '&api.Endpoints{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"kube-scheduler", GenerateName:"", Namespace:"kube-system", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimes
m0 kube-scheduler: I0603 14:41:02.243250   29837 leaderelection.go:215] sucessfully acquired lease kube-system/kube-scheduler
m0 kube-scheduler: E0603 14:41:02.263170   29837 event.go:257] Could not construct reference to: '&api.Endpoints{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"kube-scheduler", GenerateName:"", Namespace:"kube-system", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimes
m0 kube-scheduler: I0603 14:41:02.263248   29837 leaderelection.go:215] sucessfully acquired lease kube-system/kube-scheduler
m0 kube-scheduler: E0603 14:41:02.263327   29837 runtime.go:58] Recovered from panic: "close of closed channel" (close of closed channel)
m0 kube-scheduler: pkg/util/runtime/runtime.go:52
m0 kube-scheduler: pkg/util/runtime/runtime.go:40
m0 kube-scheduler: /usr/local/go/src/runtime/asm_amd64.s:472
m0 kube-scheduler: /usr/local/go/src/runtime/panic.go:426
m0 kube-scheduler: /usr/local/go/src/runtime/chan.go:300
m0 kube-scheduler: pkg/client/leaderelection/leaderelection.go:216
m0 kube-scheduler: pkg/util/wait/wait.go:86
m0 kube-scheduler: pkg/util/wait/wait.go:87
m0 kube-scheduler: pkg/util/wait/wait.go:49
m0 kube-scheduler: pkg/client/leaderelection/leaderelection.go:217
m0 kube-scheduler: pkg/client/leaderelection/leaderelection.go:175
m0 kube-scheduler: pkg/client/leaderelection/leaderelection.go:189
m0 kube-scheduler: plugin/cmd/kube-scheduler/app/server.go:157
m0 kube-scheduler: plugin/cmd/kube-scheduler/scheduler.go:47
m0 kube-scheduler: /usr/local/go/src/runtime/proc.go:188
m0 kube-scheduler: /usr/local/go/src/runtime/asm_amd64.s:1998
```

**Comment by therc** [comment]
@mikedanese

**Comment by therc** [comment]
I removed it for brevity, but those entries all come from the same PID, 29837. I haven't dug through the code yet to figure why the scheduler called the election code twice.

**Comment by mikedanese** [comment]
Looking at the code, it seems like this is a bug in wait.Until. Most recent change was https://github.com/kubernetes/kubernetes/commit/3ec25c54252568b0e3d97feffc9363c9f1e6ff1f

**Comment by timothysc** [comment]
@runcom do you have repro instructions?  

What is odd is you have b2b : "sucessfully acquired lease"

**Comment by runcom** [comment]
@timothysc ?

**Comment by timothysc** [comment]
Sorry nick misfire on autocomplete.. @therc

**Comment by timothysc** [comment]
nvmd I see the issue(s), fix in flight.

**Comment by therc** [comment]
I haven't seen this since, but thanks for fixing it!

**Comment by timothysc** [comment]
closed via #29699 and #29743
